### PR TITLE
Fix failing keep-alive timeout tests

### PR DESCRIPTION
**Pull request by cesarbs**
@halter73 @pakrym

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs [hunk 1]
```diff
@@ -91,23 +91,27 @@ public class KeepAliveTimeoutTests
         {
             using (var connection = new TestConnection(server.Port))
             {
-                for (var i = 0; i < 10; i++)
+                for (var i = 0; i < 11; i++)
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs]: 11? Why not 10?

Comment (benaadams) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs]: Goes to 11 https://www.youtube.com/watch?v=KOO5S4vxi0o :wink:

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs]: @pakrym Ha, that's not necessary. I had it in my head that the keep-alive timeout I was using for the tests was 30s, but it's 10s. With the short delay being 3s I'd need 11 iterations to make sure I was getting past the timeout.

**Comment by cesarbs** [comment]
Updated.

**Comment by cesarbs** [comment]
Travis failure is unrelated.

**Comment by cesarbs** [comment]
:watch:

**Comment by halter73** [comment]
:shipit:

**Comment by mikeharder** [comment]
@cesarbs: I thought we decided the keepalive should apply to both requests and responses, so the original test is actually better.  Is this fix temporary until the keepalive behavior can be changed?

**Comment by cesarbs** [comment]
That's right, it's temporary. The timeout implementation needs to be revisited, but this keeps the CI good for now.

**Pull request merged**